### PR TITLE
Report struct/union/array members in -var-update response 

### PR DESCRIPTION
**Pull request by anthony-kolesov**
Fixes https://github.com/lldb-tools/lldb-mi/issues/25

### CODE REVIEW

File: src/MICmdCmdVar.cpp [hunk 1]
```diff
@@ -343,6 +356,73 @@ bool CMICmdCmdVarUpdate::ParseArgs() {
...
+    CMICmnLLDBDebugSessionInfo::VariableInfoFormat_e eVarInfoFormat,
+    lldb::SBValue &value, const CMIUtilString &valueName) {
+  if (value.GetNumChildren() == 0) {
+    // Print scalar values.
+    const bool bPrintValue =
+        ((eVarInfoFormat ==
+          CMICmnLLDBDebugSessionInfo::eVariableInfoFormat_AllValues) ||
+         (eVarInfoFormat ==
+              CMICmnLLDBDebugSessionInfo::eVariableInfoFormat_SimpleValues &&
+          value.GetNumChildren() == 0));
```
Comment (tkrasnukha) [inline src/MICmdCmdVar.cpp:385]: You already checked `value.GetNumChildren() == 0` above.

File: src/MICmdCmdVar.cpp [hunk 2]
```diff
@@ -343,6 +356,73 @@ bool CMICmdCmdVarUpdate::ParseArgs() {
...
+      // -var-set-update-range. LLDB-MI doens't support the latter, though.
+      // I don't know any better way to check if member has been listed than
+      // to attempt to get the member by its name.
+      const CMIUtilString childName(
+          GetMemberName(valueName, CMICmnLLDBUtilSBValue(child).GetName(), i));
+      CMICmnLLDBDebugSessionInfoVarObj childVarObj;
+      if (!CMICmnLLDBDebugSessionInfoVarObj::VarObjGet(childName,
+                                                       childVarObj)) {
+        continue;
+      }
```
Comment (tkrasnukha) [inline src/MICmdCmdVar.cpp:417]: Unnecessary braces around "continue".

**Comment by anthony-kolesov** [comment]
I've changed children-value handling, because a pointer to a structure is, apparently, considered a structure with children but LLDB, but no children are printed. But Eclipse on it's own handles pointers as complex structures, rather than just addresses. So in the end, this patch handles local variables, but doesn't handle pointers. Visually it looks like this (`s.f3.parent` points to s, so in theory it's fields should have been udpated as well):

image

**Comment by anthony-kolesov** [comment]
I think this can be merged, I don't plan any more fixes.

**Pull request merged**